react/features/facial-recognition/actions.js

```diff
@@ -52,80 +59,97 @@ let duplicateConsecutiveExpressions = 0;
 /**
  * Variable that keeps the interval for sending expressions to webhook.
  */
-let sendInterval;
+let webhookSendInterval;
+
+/**
+ * Variable that keeps the interval for detecting faces in a frame.
+ */
+let detectionInterval;
 
 /**
  * Loads the worker that predicts the facial expression.
  *
  * @returns {void}
  */
 export function loadWorker() {
-    return function(dispatch: Function) {
-        if (!window.Worker) {
-            logger.warn('Browser does not support web workers');
+    return function(dispatch: Function, getState: Function) {
+        if (navigator.product === 'ReactNative') {
+            logger.warn('Unsupported environment for face recognition');
 
             return;
         }
 
-        const baseUrl = `${getBaseUrl()}/libs/`;
+        const baseUrl = `${getBaseUrl()}libs/`;
         let workerUrl = `${baseUrl}facial-expressions-worker.min.js`;
 
         const workerBlob = new Blob([ `importScripts("${workerUrl}");` ], { type: 'application/javascript' });
 
         workerUrl = window.URL.createObjectURL(workerBlob);
-        worker = new Worker(workerUrl, { name: 'Facial Expression Worker' });
+        worker = new Worker(workerUrl, { name: 'Face Recognition Worker' });
         worker.onmessage = function(e: Object) {
             const { type, value } = e.data;
 
-            // receives a message indicating what type of backend tfjs decided to use.
-            // it is received after as a response to the first message sent to the worker.
-            if (type === INTERVAL_MESSAGE) {
-                value && dispatch(setDetectionTimeInterval(value));
-            }
-
-            // receives a message with the predicted facial expression.
-            if (type === FACIAL_EXPRESSION_MESSAGE) {
-                sendDataToWorker(worker, imageCapture);
-                if (!value) {
-                    return;
-                }
+            switch (type) {
+            case FACIAL_EXPRESSION_MESSAGE: {
```

Comment:
Thinking about future extensibility, I think it would be a good idea to use a single type of message with an object as the payload. Each of the landmarks (emotion / experssion and face box currently) would be a key so we process them if we find them.

Reply:
arguably, processing these messages take different code paths, and you'd only replace the switch with if statements that check for the respective keys in order to take the appropriate execution path.